go/ms/sigcomm/mssighandlers.go

```diff
@@ -0,0 +1,127 @@
+package sigcomm
+
+import (
+	"context"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/scionproto/scion/go/lib/ctrl"
+	"github.com/scionproto/scion/go/lib/ctrl/ms_mgmt"
+	"github.com/scionproto/scion/go/lib/infra"
+	"github.com/scionproto/scion/go/lib/infra/messenger"
+	"github.com/scionproto/scion/go/lib/log"
+	"github.com/scionproto/scion/go/lib/snet"
+	"github.com/scionproto/scion/go/ms/internal/mscrypto"
+	"github.com/scionproto/scion/go/ms/internal/msmsgr"
+	"github.com/scionproto/scion/go/ms/internal/sqlite"
+	"github.com/scionproto/scion/go/ms/internal/validator"
+	"github.com/scionproto/scion/go/pkg/trust"
+	"github.com/scionproto/scion/go/proto"
+)
+
+const (
+	/* //TODO (supraja): this should be configurable and related
+	to other timestamp values to be added later. See doc*/
+	max_ms_as_add_time = 10 //time is in minutes.
+)
+
+//ASActionHandler is a handler for messages from SIG
+type ASActionHandler struct {
+}
+
+//Handle handles an AS Action request from a SIG
+//The SIG sends ms_mgmt.Pld with ms_mgmt.ASMapEntry. This handler verifies signatures,
+//the ownership of the prefixes using RPKI and adds it to the
+//new_entries database for further processing. It also sends back a ms_mgmt.MSRepToken
+//to the SIG
+func (a ASActionHandler) Handle(r *infra.Request) *infra.HandlerResult {
+	log.Info("Entering: ASActionHandler.Handle")
+	ctx := r.Context()
+	requester := r.Peer.(*snet.UDPAddr)
+	m := r.FullMessage.(*ctrl.SignedPld)
+	e := mscrypto.MSEngine{Msgr: msmsgr.Msgr, IA: msmsgr.IA}
+	verifier := trust.Verifier{BoundIA: requester.IA, Engine: e}
+	err := verifier.Verify(context.Background(), m.Blob, m.Sign)
+	rw, _ := infra.ResponseWriterFromContext(r.Context())
+	sendAck := messenger.SendAckHelper(ctx, rw)
+	if err != nil {
+		log.Error("Certificate verification failed!")
+		sendAck(proto.Ack_ErrCode_reject, err.Error())
+		return nil
+	}
+
+	//Source IA validated here, make sure the source AS and the ASMap have the same address
+	asMapEntry := r.Message.(*ms_mgmt.ASMapEntry)
+	if requester.IA.String() != asMapEntry.Ia {
+		log.Error("Invalid AS in map")
+		sendAck(proto.Ack_ErrCode_reject, "Invalid AS in map")
+		return nil
+	}
+	//All good. Source IA is the IA in the asMap as well.
+
+	//Now validate the AS-IP mapping using an rpkivalidator
+	//Do RPKI validation with a shell script for now
+	//For now, the validator should take 2 arguments,
+	//ASN and prefix and return "valid" if the mapping is valid
+	cmdStr := validator.Path + " " + requester.IA.A.String() + " " + asMapEntry.Ip[0]
+	cmd := exec.Command("/bin/sh", "-c", cmdStr)
```

Comment:
>	cmdStr := validator.Path + " " + requester.IA.A.String() + " " + asMapEntry.Ip[0]
>	cmd := exec.Command("/bin/sh", "-c", cmdStr)

Don't build the command string like that, do it like this https://golang.org/pkg/os/exec/#example_Command, i.e. directly as part of the arguments to Command.

Reply:
Done!